smartsim/_core/mli/infrastructure/worker/worker.py

```diff
@@ -71,11 +74,15 @@ def __init__(
         outputs: t.Optional[t.Collection[t.Any]] = None,
         output_keys: t.Optional[t.Collection[str]] = None,
         failed: bool = False,
```

Comment:
Should we dump `failed` and just use the value of `status` instead?

Reply:
I won't lie, keeping  a `failed` parameter makes the if checks I make throughout a lot easier.

Reply:
This makes me want to add a n `in progress` enum so I don't immediately set it to `complete`. I think I'll do that.